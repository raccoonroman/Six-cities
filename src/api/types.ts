import { OfferCategory, Location } from '@/types';

export interface AuthInfoRaw {
  avatar_url: string;
  email: string;
  id: number;
  is_pro: boolean;
  name: string;
}

export interface User {
  email: string;
  password: string;
}

export interface OfferRaw {
  id: number;
  title: string;
  preview_image: string;
  price: number;
  rating: number;
  type: OfferCategory;
  bedrooms: number;
  max_adults: number;
  is_favorite: boolean;
  is_premium: boolean;
  location: Location;
  city: {
    name: string;
    location: Location;
  };
  description: string;
  goods: string[];
  host: {
    avatar_url: string;
    id: number;
    is_pro: boolean;
    name: string;
  };
  images: string[];
}

export interface CommentRaw {
  id: number;
  comment: string;
  date: string;
  rating: number;
  user: {
    avatar_url: string;
    id: number;
    is_pro: boolean;
    name: string;
  };
}

export interface CommentPost {
  comment: string;
  rating: number;
}
