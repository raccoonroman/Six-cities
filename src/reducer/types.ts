import { OfferRaw } from '@/api/types';
import { ActionType } from '@/const';

interface LoadOffersAction {
  type: typeof ActionType.LOAD_OFFERS
  payload: OfferRaw[]
}

interface UpdateOfferAction {
  type: typeof ActionType.UPDATE_OFFER
  payload: OfferRaw
}

interface SetEmailAction {
  type: typeof ActionType.SET_EMAIL
  payload: string
}

export type offersAction = LoadOffersAction | UpdateOfferAction;
export type userDataAction = SetEmailAction;
