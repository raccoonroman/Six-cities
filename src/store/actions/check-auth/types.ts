enum CheckAuth {
  PENDING = 'CHECK_AUTH/PENDING',
  RESOLVE = 'CHECK_AUTH/RESOLVE',
  REJECT = 'CHECK_AUTH/REJECT',
}

export default CheckAuth;
