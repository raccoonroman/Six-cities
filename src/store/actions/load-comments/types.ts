enum LoadComments {
  PENDING = 'LOAD_COMMENTS/PENDING',
  RESOLVE = 'LOAD_COMMENTS/RESOLVE',
  REJECT = 'LOAD_COMMENTS/REJECT',
}

export default LoadComments;
