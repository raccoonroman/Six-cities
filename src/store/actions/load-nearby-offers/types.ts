enum LoadNearbyOffers {
  PENDING = 'LOAD_NEARBY_OFFERS/PENDING',
  RESOLVE = 'LOAD_NEARBY_OFFERS/RESOLVE',
  REJECT = 'LOAD_NEARBY_OFFERS/REJECT',
}

export default LoadNearbyOffers;
