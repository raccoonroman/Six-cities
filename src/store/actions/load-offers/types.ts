enum LoadOffers {
  PENDING = 'LOAD_OFFERS/PENDING',
  RESOLVE = 'LOAD_OFFERS/RESOLVE',
  REJECT = 'LOAD_OFFERS/REJECT',
}

export default LoadOffers;
