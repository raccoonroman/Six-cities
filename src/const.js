export const DEFAULT_CITIES = [`Paris`, `Cologne`, `Brussels`, `Amsterdam`, `Hamburg`, `Dusseldorf`];

export const SortType = {
  POPULAR: `Popular`,
  PRICE_LOW_TO_HIGH: `Price: low to high`,
  PRICE_HIGH_TO_LOW: `Price: high to low`,
  TOP_RATED_FIRST: `Top rated first`,
};

export const CardType = {
  CITY: `city`,
  NEAR: `near`,
  FAVORITE: `favorite`,
};

export const ActionType = {
  LOAD_OFFERS: `LOAD_OFFERS`,
  UPDATE_OFFER: `UPDATE_OFFER`,
  SET_CITY: `SET_CITY`,
  REQUIRED_AUTHORIZATION: `REQUIRED_AUTHORIZATION`,
  SET_EMAIL: `SET_EMAIL`,
  LOAD_COMMENTS: `LOAD_COMMENTS`,
  LOAD_NEARBY_OFFERS: `LOAD_NEARBY_OFFERS`,
};

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const AppRoute = {
  ROOT: `/`,
  LOGIN: `/login`,
  FAVORITES: `/favorites`,
  OFFER: `/offer`,
};
