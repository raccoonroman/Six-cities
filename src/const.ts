export const DEFAULT_CITIES: string[] = ['Paris', 'Cologne', 'Brussels', 'Amsterdam', 'Hamburg', 'Dusseldorf'];

export enum SortType {
  POPULAR = 'Popular',
  PRICE_LOW_TO_HIGH = 'Price: low to high',
  PRICE_HIGH_TO_LOW = 'Price: high to low',
  TOP_RATED_FIRST = 'Top rated first',
}

export enum CardType {
  CITY = 'city',
  NEAR = 'near',
  FAVORITE = 'favorite',
}

export enum ActionType {
  LOAD_OFFERS = 'LOAD_OFFERS',
  UPDATE_OFFER = 'UPDATE_OFFER',
  SET_CITY = 'SET_CITY',
  REQUIRED_AUTHORIZATION = 'REQUIRED_AUTHORIZATION',
  SET_EMAIL = 'SET_EMAIL',
  LOAD_COMMENTS = 'LOAD_COMMENTS',
  LOAD_NEARBY_OFFERS = 'LOAD_NEARBY_OFFERS',
}

export enum AuthorizationStatus {
  AUTH = 'AUTH',
  NO_AUTH = 'NO_AUTH',
}

export enum AppRoute {
  ROOT = '/',
  LOGIN = '/login',
  FAVORITES = '/favorites',
  OFFER = '/offer',
}

export enum MapType {
  HOVERED_ACTIVE_OFFER = 'HOVERED_ACTIVE_OFFER',
  STATIC_ACTIVE_OFFER = 'STATIC_ACTIVE_OFFER',
}
