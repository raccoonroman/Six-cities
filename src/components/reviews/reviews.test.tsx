import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import {AuthorizationStatus} from '../../const';
import {CommentRaw} from '../../types';
import Reviews from './reviews';


const mockStore = configureStore([]);

const commentsByOffer: CommentRaw[] = [
  {
    'id': 0,
    'comment': `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.`,
    'date': `2011-10-05T14:48:00.000Z`,
    'rating': 5,
    'user': {
      'avatar_url': `https://i.picsum.photos/id/220/200/200.jpg`,
      'id': 0,
      'is_pro': true,
      'name': `Rachel`,
    }
  },
  {
    'id': 1,
    'comment': `Lorem ipsum dolor sit amet`,
    'date': `2019-12-08T12:18:10.000Z`,
    'rating': 1,
    'user': {
      'avatar_url': `https://i.picsum.photos/id/221/200/200.jpg`,
      'id': 1,
      'is_pro': false,
      'name': `Monica`,
    }
  },
];


describe(`Render <Reviews />`, () => {
  it(`When user is not authorized`, () => {
    const store = mockStore({
      authorization: {
        authorizationStatus: AuthorizationStatus.NO_AUTH,
      },
      userData: {
        email: ``,
      },
      commentsByOffer,
    });

    const tree = renderer
    .create(
        <Provider store={store}>
          <Reviews offerId={0} />
        </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`When user is authorized`, () => {
    const store = mockStore({
      authorization: {
        authorizationStatus: AuthorizationStatus.AUTH,
      },
      userData: {
        email: `name@gmail.com`,
      },
      commentsByOffer,
    });

    const tree = renderer
    .create(
        <Provider store={store}>
          <Reviews offerId={0} />
        </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
