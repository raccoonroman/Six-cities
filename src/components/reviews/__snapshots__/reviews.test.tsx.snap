// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render <Reviews /> When user is authorized 1`] = `
<section
  className="property__reviews reviews"
>
  <h2
    className="reviews__title"
  >
    Reviews ·
    <span
      className="reviews__amount"
    >
      2
    </span>
  </h2>
  <ul
    className="reviews__list"
  >
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Monica"
            className="reviews__avatar user__avatar"
            height="54"
            src="https://i.picsum.photos/id/221/200/200.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Monica
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "20%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          Lorem ipsum dolor sit amet
        </p>
        <time
          className="reviews__time"
          dateTime="2019-12-08"
        >
          December 2019
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Rachel"
            className="reviews__avatar user__avatar"
            height="54"
            src="https://i.picsum.photos/id/220/200/200.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Rachel
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "100%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
        </p>
        <time
          className="reviews__time"
          dateTime="2011-10-05"
        >
          October 2011
        </time>
      </div>
    </li>
  </ul>
  <form
    action="#"
    className="reviews__form form"
    method="post"
    onSubmit={[Function]}
  >
    <label
      className="reviews__label form__label"
      htmlFor="review"
    >
      Your review
    </label>
    <div
      className="reviews__rating-form form__rating"
    >
      <input
        checked={false}
        className="form__rating-input visually-hidden"
        disabled={false}
        id="5-stars"
        name="rating"
        onChange={[Function]}
        type="radio"
        value={5}
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="5-stars"
        title="perfect"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
      <input
        checked={false}
        className="form__rating-input visually-hidden"
        disabled={false}
        id="4-stars"
        name="rating"
        onChange={[Function]}
        type="radio"
        value={4}
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="4-stars"
        title="good"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
      <input
        checked={false}
        className="form__rating-input visually-hidden"
        disabled={false}
        id="3-stars"
        name="rating"
        onChange={[Function]}
        type="radio"
        value={3}
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="3-stars"
        title="not bad"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
      <input
        checked={false}
        className="form__rating-input visually-hidden"
        disabled={false}
        id="2-stars"
        name="rating"
        onChange={[Function]}
        type="radio"
        value={2}
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="2-stars"
        title="badly"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
      <input
        checked={false}
        className="form__rating-input visually-hidden"
        disabled={false}
        id="1-stars"
        name="rating"
        onChange={[Function]}
        type="radio"
        value={1}
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="1-stars"
        title="terribly"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
    </div>
    <textarea
      className="reviews__textarea form__textarea"
      disabled={false}
      id="review"
      name="review"
      onChange={[Function]}
      placeholder="Tell how was your stay, what you like and what can be improved"
      value=""
    />
    <div
      className="reviews__button-wrapper"
    >
      <p
        className="reviews__help"
      >
        To submit review please make sure to set 
        <span
          className="reviews__star"
        >
          rating
        </span>
         and describe your stay with 
        <b
          className="reviews__text-amount"
        >
          minimum 50 and maximum 300 characters
        </b>
        .
      </p>
      <button
        className="reviews__submit form__submit button"
        disabled={true}
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</section>
`;

exports[`Render <Reviews /> When user is not authorized 1`] = `
<section
  className="property__reviews reviews"
>
  <h2
    className="reviews__title"
  >
    Reviews ·
    <span
      className="reviews__amount"
    >
      2
    </span>
  </h2>
  <ul
    className="reviews__list"
  >
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Monica"
            className="reviews__avatar user__avatar"
            height="54"
            src="https://i.picsum.photos/id/221/200/200.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Monica
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "20%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          Lorem ipsum dolor sit amet
        </p>
        <time
          className="reviews__time"
          dateTime="2019-12-08"
        >
          December 2019
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Rachel"
            className="reviews__avatar user__avatar"
            height="54"
            src="https://i.picsum.photos/id/220/200/200.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Rachel
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "100%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
        </p>
        <time
          className="reviews__time"
          dateTime="2011-10-05"
        >
          October 2011
        </time>
      </div>
    </li>
  </ul>
</section>
`;
